src/Core/Config/Util/ConfigFileManager.php

```diff
@@ -200,8 +214,13 @@ public function saveData(Cache $config)
 			throw new ConfigFileException('config source cannot get encoded');
 		}
 
-		if (!file_put_contents($this->configDir . '/' . self::CONFIG_DATA_FILE, $encodedData)) {
-			throw new ConfigFileException(sprintf('Cannot save data to file %s/%s', $this->configDir, self::CONFIG_DATA_FILE));
+		$configStream = fopen($this->configDir . '/' . self::CONFIG_DATA_FILE, 'w+');
+
+		if (flock($configStream, LOCK_EX)) {
+			ftruncate($configStream, 0);
+			fwrite($configStream, $encodedData);
+			fflush($configStream);
+			flock($configStream, LOCK_UN);
```

Comment:
No reading is done with this stream, so you can simplify to `w`, and `w` already truncates the file to 0 length per documentation: https://www.php.net/manual/en/function.fopen.php
```suggestion
		$configStream = fopen($this->configDir . '/' . self::CONFIG_DATA_FILE, 'w');

		if (flock($configStream, LOCK_EX)) {
			fwrite($configStream, $encodedData);
			fflush($configStream);
			flock($configStream, LOCK_UN);
```